src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -172,8 +172,9 @@ export class KernelProcess implements IKernelProcess {
         try {
             await Promise.race([
                 tcpPortUsed.waitUntilUsed(this.connection.hb_port, 200, timeout),
+                deferred.promise,
```

Comment:
Used a promise instead of a cancellation token, else we lose all of the error information.